user-app-test/src/modules/sns/sns.provider.ts

```diff
@@ -0,0 +1,16 @@
+// sns.provider.ts
+import { ConfigService } from '@nestjs/config';
+import * as AWS from 'aws-sdk';
+
+export const SnsProvider = {
+  provide: 'SNS',
+  useFactory: (configService: ConfigService) => {
+    return new AWS.SNS({
+      endpoint: configService.get<string>('SNS_ENDPOINT'),
+      region: configService.get<string>('AWS_REGION'),
+      accessKeyId: configService.get<string>('AWS_ACCESS_KEY_ID'),
+      secretAccessKey: configService.get<string>('AWS_SECRET_ACCESS_KEY'),
+    });
+  },
+  inject: [ConfigService],
+};
```

Comment:
nitpick:
```suggestion
export const AWS_SNS_PROVIDER = Symbol('SNS');

export const SnsProvider = {
  provide: AWS_SNS_PROVIDER,
  useFactory: (configService: ConfigService) => {
    return new AWS.SNS({
      endpoint: configService.get<string>('SNS_ENDPOINT'),
      region: configService.get<string>('AWS_REGION'),
      accessKeyId: configService.get<string>('AWS_ACCESS_KEY_ID'),
      secretAccessKey: configService.get<string>('AWS_SECRET_ACCESS_KEY'),
    });
  },
  inject: [ConfigService],
};
```

тогда мы и другие всегда точно будем знать какой ключ юзать и он будет уникальным, даже если кто-то другой сделает такой же